src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -254,7 +267,33 @@ private static IEnumerable<string> GetDiagnosticsAnalyzerSupportedLanguages(PEMo
             // first supported language and an array parameter for addition supported languages.
             // Parse the argument blob to extract the languages.
             BlobReader argsReader = peModule.GetMemoryReaderOrThrow(peModule.GetCustomAttributeValueOrThrow(customAttrHandle));
+            return ReadLanguagesFromAttribute(ref argsReader);
+        }
 
+        private static bool IsGeneratorAttribute(PEModule peModule, CustomAttributeHandle customAttrHandle)
```

Comment:
Could this be a local function?

Reply:
I refactored out the extensions logic to make it simpler and not need it.